flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractFetcher.java

```diff
@@ -507,7 +507,7 @@ private void updateMinPunctuatedWatermark(Watermark nextWatermark) {
 			SerializedValue<AssignerWithPunctuatedWatermarks<T>> watermarksPunctuated,
 			ClassLoader userCodeClassLoader) throws IOException, ClassNotFoundException {
 
-		List<KafkaTopicPartitionState<KPH>> partitionStates = new LinkedList<>();
+		List<KafkaTopicPartitionState<KPH>> partitionStates = new CopyOnWriteArrayList<>();
```

Comment:
Would be nice to have a comment on why we need to use a `CopyOnWriteArrayList`

Reply:
Yes you are right. A question: is it allowed to specify a link on the issue comment where it was decided to use CopyOnWriteArrayList? Or is it better to have explanation in a comment only?

Reply:
explanation can be made with a comment.
No need to link to issue comment.